reframe/core/pipeline.py

```diff
@@ -1436,10 +1436,7 @@ def check_performance(self):
 
                 for var in variables:
                     name, unit = var
-                    ref_tuple = (0, None, None)
-                    if unit:
-                        ref_tuple += (unit,)
-
+                    ref_tuple = (0, None, None, unit or '(no unit specified)')
```

Comment:
I would keep the unit as `None`. The JSON schema allows it to be so.

```suggestion
                    ref_tuple = (0, None, None, unit)
```